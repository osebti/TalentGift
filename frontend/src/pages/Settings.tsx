import Button from "@/components/Button";
import PageTitle from "@/components/PageTitle";

function DeleteOrgButton() {
  return (
    <Button
      id="delete-org"
      label="Delete Organization"
      href="/dashboard/delete-org"
      navigate
      style="btn-secondary rounded-2xl"
    />
  );
}

export default function Settings() {
  return (
    <div id="settings-outer-container" className="px-4 md:px-8">
      <PageTitle id="settings" title="Settings" style="mb-4" />
      <DeleteOrgButton />
    </div>
  );
}
