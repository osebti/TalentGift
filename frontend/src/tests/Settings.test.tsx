import { cleanup, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Settings from "@/pages/Settings";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

function renderSettingsPage() {
  return (
    render(
      <BrowserRouter>
        <Settings/>
      </BrowserRouter>
    )
  );
}

describe("Rendering the Settings Page", () => {

  it("renders correctly", () => {
    const page = renderSettingsPage();
    expect(page).toMatchSnapshot();
  });

  it("renders the delete organization button", () => {
    const { queryByTestId } = renderSettingsPage();
    expect(queryByTestId("delete-org-button")).not.toBeNull();
  });

});

describe("Deleting the Organization", () => {
  
  it("directs to organization deletion page", () => {
    // click on the delete organization button
    const { queryByTestId } = renderSettingsPage();
    const deleteOrgButton = queryByTestId("delete-org-button");
    if (deleteOrgButton != null) {
      fireEvent.click(deleteOrgButton);
    }
    // check if the user is redirected to the delete organization page
    expect(global.window.location.href).toContain("/delete-org");
  });

});