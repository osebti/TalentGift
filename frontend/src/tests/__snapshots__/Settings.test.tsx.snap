// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the Settings Page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-4 md:px-8"
        id="settings-outer-container"
      >
        <div
          class="w-full font-semibold text-2xl md:text-3xl lg:text-4xl text-black mb-4"
          data-testid="settings-page-title"
          id="settings-page-title"
        >
          Settings
        </div>
        <a
          class="btn btn-md capitalize font-normal text-white text-sm md:text-base btn-secondary rounded-2xl"
          data-testid="delete-org-button"
          href="/dashboard/delete-org"
          id="delete-org-button"
        >
          Delete Organization
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-4 md:px-8"
      id="settings-outer-container"
    >
      <div
        class="w-full font-semibold text-2xl md:text-3xl lg:text-4xl text-black mb-4"
        data-testid="settings-page-title"
        id="settings-page-title"
      >
        Settings
      </div>
      <a
        class="btn btn-md capitalize font-normal text-white text-sm md:text-base btn-secondary rounded-2xl"
        data-testid="delete-org-button"
        href="/dashboard/delete-org"
        id="delete-org-button"
      >
        Delete Organization
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
